src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/utils/TaskMandatoryFieldsValidator.java

```diff
@@ -0,0 +1,139 @@
+    public static final LDValue MANDATORY_FIELD_CHECK_FLAG_VARIANT = LDValue.of("jurisdictions");
+
+    /**
+     * Constructor for TaskMandatoryFieldsValidator.
+     *
+     * @param launchDarklyFeatureFlagProvider the LaunchDarkly feature flag provider
+     * @param taskMandatoryFieldCheckEnabled  flag to enable or disable mandatory field check
+     * @param taskMandatoryFields             list of mandatory fields to be checked
+     */
+    @Autowired
+    public TaskMandatoryFieldsValidator(LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider,
+                                        @Value("${config.taskMandatoryFieldCheckEnabled}")
+                                        Boolean taskMandatoryFieldCheckEnabled,
+                                        @Value("${config.taskMandatoryFields}") List<String> taskMandatoryFields) {
+        this.launchDarklyFeatureFlagProvider = launchDarklyFeatureFlagProvider;
+        this.isMandatoryFieldCheckEnabled = taskMandatoryFieldCheckEnabled;
+        this.taskMandatoryFields = taskMandatoryFields;
+    }
+
+    /**
+     * Validates the mandatory fields of a task.
+     *
+     * @param task the task to be validated
+     */
+    public void validate(TaskResource task) {
+        log.info("Validating mandatory fields for task {}", task.getTaskId());
+
+        if (isMandatoryFieldCheckEnabled) {
+            LDValue mandatoryFieldCheckEnabledServices = launchDarklyFeatureFlagProvider.getJsonValue(
+                FeatureFlag.WA_MANDATORY_FIELD_CHECK,
+                "ccd-case-disposer",
+                "ccd-case-disposer@hmcts.net",
+                LDValue.of("{\"jurisdictions\": []}")
+            );
+
+            if (mandatoryFieldCheckEnabledServices == null) {
+                log.warn("Mandatory field check enabled services is null, skipping validation.");
+                return;
+            }
+
+            JsonNode excludedJurisdictionsArray = parseJson(mandatoryFieldCheckEnabledServices.toJsonString());
+            if (excludedJurisdictionsArray == null) {
+                log.warn("Excluded jurisdictions array is null, skipping jurisdiction check.");
+                validateTaskMandatoryFields(task);
+                return;
+            }
+
+            if (!isJurisdictionExcluded(task, excludedJurisdictionsArray)) {
+                validateTaskMandatoryFields(task);
+            }
+        }
+    }
+
+    /**
+     * Checks if the task's jurisdiction is excluded.
+     *
+     * @param task the task to be checked
+     * @param excludedJurisdictionsArray the array of excluded jurisdictions
+     * @return true if the jurisdiction is excluded, false otherwise
+     */
+    private boolean isJurisdictionExcluded(TaskResource task, JsonNode excludedJurisdictionsArray) {
+        for (JsonNode jurisdiction : excludedJurisdictionsArray) {
+            if (jurisdiction.asText().equals(task.getJurisdiction())) {
+                return true;
```

Comment:
Possibly a log of an excluded jurisdiction would be helpful for debug.  Over time there should be no exclusions would be good to know if for instance we end up with some new empty Role Categories how that happened.